MetaMask/metamask-extension
# fix: add img role to avatar for improved screen reader experience

## **Description**

To improve the accessibility of our Avatar components, especially when a screen reader is used, we propose adding the role="img" attribute to these components. This enhancement aims to prevent screen readers from reading out the fallback letter, providing a more meaningful and contextually appropriate description. For instance, when tabbing through networks in the extension, the screen reader should announce "S, Sepolia" instead of just "S". This change will ensure our UI is more accessible and user-friendly for individuals relying on screen readers.



[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/23832?quickstart=1)

## **Related issues**

Fixes: #23251 

## **Manual testing steps**

1. Turn on your computers screen reader (mac: CMD + F5)
2. Run local build of extension 
3. Tab to a component that uses an avatar component (e.g. see screenshot for example component) 

## **Screenshots/Recordings**



### **Before**

https://github.com/MetaMask/metamask-extension/assets/8112138/ec6ab28f-8fbf-4cd2-b0ea-787f82d7bcae



### **After**

https://github.com/MetaMask/metamask-extension/assets/26469696/18b6d1b8-7268-49f3-92b4-4ec10ef382bc




## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
